native/cocos/platform/openharmony/FileUtils-OpenHarmony.cpp

```diff
@@ -0,0 +1,306 @@
+#include "cocos/platform/openharmony/FileUtils-OpenHarmony.h"
+#include <hilog/log.h>
+#include <sys/stat.h>
+#include <cstdio>
+#include <regex>
+#include <string>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+
+#include "cocos/base/Log.h"
+
+#include <sys/syscall.h>
+#include <unistd.h>
+#include "bindings/jswrapper/napi/HelperMacros.h"
+
+#define ASSETS_FOLDER_WRITEABLE_PATH "/data/accounts/account_0/applications/ohos.example.xcomponent1/ohos.example.xcomponent1/writeable_path"
+
+#ifndef JCLS_HELPER
+    #define JCLS_HELPER "com/cocos/lib/CocosHelper"
+#endif
+#include "rawfile/raw_file_manager.h"
+
+namespace cc {
+
+namespace {
+
+std::string rawfilePrefix = "entry/resources/rawfile/";
+
+void printRawfiles(NativeResourceManager *mgr, const std::string &path) {
+//    auto *file = OpenRawFile(mgr, path.c_str());
+//    if (file) {
+//        HILOG_DEBUG(LOG_APP, "PrintRawfile %{public}s", path.c_str());
+//        return;
+//    }
+//
+//    RawDir *dir = OpenRawDir(mgr, path.c_str());
+//    if (dir) {
+//        auto fileCnt = GetRawFileCount(dir);
+//        for (auto i = 0; i < fileCnt; i++) {
+//            std::string subFile  = GetRawFileName(dir, i);
+//            auto        newPath  = path + "/" + subFile; // NOLINT
+//            auto        debugPtr = newPath.c_str();
+//            HILOG_ERROR(LOG_APP, " find path %{public}s", newPath.c_str());
+//            printRawfiles(mgr, newPath);
+//        }
+//    } else {
+//        HILOG_ERROR(LOG_APP, "Invalidate path %{public}s", path.c_str());
+//    }
+}
+} // namespace
+
+NativeResourceManager* FileUtilsOpenHarmony::_nativeResourceManager = nullptr;
+
+bool FileUtilsOpenHarmony::initResourceManager(napi_env env, napi_value param) {
+    _nativeResourceManager = OH_ResourceManager_InitNativeResourceManager(env, param);
+    LOGE("cocos qgh  initResourceManager %{public}p", _nativeResourceManager);
+    return true;
+}
+
+FileUtils::Status FileUtilsOpenHarmony::getRawFileDescriptor(const std::string &filename,RawFileDescriptor& descriptor) {
+    if (filename.empty()) {
+        return FileUtils::Status::NOT_EXISTS;
```

Comment:
`PATH_INCORRECT` instead of `NOT_EXISTS` ?

Reply:
![image](https://user-images.githubusercontent.com/5205637/177950355-f0231de3-6862-48ee-838c-1ee8bd969fd2.png)
There seems to be no such definition？

Reply:
yes, the status is not complete